components/Feature/VulnerabilitiesGrid/index.js

```diff
@@ -1,19 +1,42 @@
-import { useState } from 'react';
+import { useState, useMemo } from 'react';
 import {
   Accordion,
   AccordionItem,
   Checkbox,
   CheckboxList,
   TextInput
 } from 'components/Form';
+import ResourceCard from './ResourceCard';
 import groups from './grid.json';
 
-const VulnerabilitiesGrid = ({ onUpdate }) => {
+function createLookup(resources) {
+  const lookup = new Map();
+
+  groups.forEach(group => {
+    const { name, assets, vulnerabilities } = group;
+    const values = new Set();
```

Comment:
This allows us to get `O(1)` lookups rather than looping through arrays each time we want to check if something is inside of a group.